src/components/ProposalForm.tsx

```diff
@@ -45,31 +85,47 @@ export default function ProposalForm() {
           </CardDescription>
         </CardHeader>
         <CardContent>
-          <form className="space-y-4">
+          <form className="space-y-4" id="proposalForm" onSubmit={onSubmit}>
             <div className="grid grid-cols-2 gap-4">
               <div className="space-y-2">
                 <Label htmlFor="name">Name</Label>
-                <Input id="name" placeholder="Your Name" required />
+                <Input
+                  id="name"
+                  placeholder="Your Name"
+                  type="text"
+                  name="authorName"
+                  required
+                  minLength={6}
+                />
               </div>
-              <div className="space-y-2">
+              <div id="email-field" className="space-y-2">
                 <Label htmlFor="email">Email</Label>
                 <Input
                   id="email"
+                  name="authorEmail"
                   placeholder="you@example.com"
                   required
                   type="email"
+                  pattern=".+@.+"
                 />
               </div>
             </div>
             <div className="space-y-2">
               <Label htmlFor="title">Title</Label>
-              <Input id="title" placeholder="Proposal Title" required />
+              <Input
+                id="title"
+                name="title"
+                placeholder="Proposal Title"
+                required
+                minLength={6}
+                type="text"
+              />
             </div>
             <div className="space-y-2">
               <Label htmlFor="type">Type</Label>
-              <Select>
+              <Select defaultValue="topic" defaultOpen={false}>
```

Comment:
ISSUE: Looks like I forgot a `name` attribute for this field
ISSUE: I think this field needs to be required
SUGGESTION: I actually think the UX is a little nicer if we don't set a default value for this element. 
```suggestion
              <Select required name="type" defaultOpen={false}>
```